Add iter_lines Method to FastHttpSession Class

MasterKey-Pro opened the pull request:
1、Add iter_lines Method to FastHttpSession Class
2、fix: Remove trailing whitespace from blank line in fasthttp.py to comply with PEP 8

Fixes #3018

cyberw commented:
Looks good, but can you add a test case too? (in test_fasthttp.py)

MasterKey-Pro commented:
The lack of an endpoint that can return streaming data or a simulated response might be the reason for the test failure.

cyberw commented:
Can you use the /streaming/<int> endpoint defined here? https://github.com/locustio/locust/blob/81116b20cdc097f8da8f0686c62c22946381254a/locust/test/testcases.py#L119

MasterKey-Pro commented:
> 您可以使用此处定义的 /streaming/:int: 端点吗？
> 
> https://github.com/locustio/locust/blob/81116b20cdc097f8da8f0686c62c22946381254a/locust/test/testcases.py#L119

I have just completed the implementation of the iter_lines method in FastHttpSession and successfully passed all related tests. I look forward to your review and feedback on this part of the work.

cyberw commented:
Looks nice! (sorry for the lack of activity)

Does it make sense to have a parameter for method? (it can still default to GET)

Can the test also ensure that a the request is logged? You can look at the other tests for examples.

cyberw commented:
Ping @MasterKey-Pro :)

MasterKey-Pro commented:
@cyberw Hello, apologies for the delay in responding as I was on an extended vacation. Regarding the changes, the main updates are as follows:

Added a method parameter to allow the caller to customize the HTTP method (default is still GET);
Used assert_called_once_with to ensure the request method is called with the correct parameters;
Removed the logging check since the focus was on verifying the correct HTTP method, which was handled using assert_called_once_with.

cyberw commented:
Awesome! I'll make sure it gets visible in the documentation too.

MasterKey-Pro commented:
@cyberw  I've always wanted to express my gratitude, and I'm glad I finally have the chance. It’s an honor to be part of this, which has brought me great involvement and a sense of accomplishment. Thank you and the team for your hard work. Locust has helped me grow my skills in the competitive job market, while also creating a better future for me and my family. Your dedication to maintaining Locust has impacted many professionals, giving us more opportunities and hope. I wish you and the team continued success, and for Locust to keep growing
